store/src/java-test/com/zimbra/cs/service/account/GetInfoTest.java

```diff
@@ -0,0 +1,94 @@
+        Provisioning.setInstance(prov);
+        server = Provisioning.getInstance().getLocalServer();
+    }
+
+    private void setAttributes(String[] urls) {
+        Map<String, Object> map = new HashMap<>();
+        map.put("zimbraSpellCheckURL", urls);
+        server.setAttrs(map);
+    }
+
+    private void setAttributes(String[] servicesEnabled, String[] servicesInstalled, String[] urls) {
+        Map<String, Object> map = new HashMap<>();
+        map.put("zimbraSpellCheckURL", urls);
+        map.put("zimbraServiceEnabled", servicesEnabled);
+        map.put("zimbraServiceInstalled", servicesInstalled);
+        server.setAttrs(map);
+    }
+
+    /**
+     * spellcheck should return true when zimbraSpellCheckURL on the current server returns non-empty list of urls.
+     */
+    @Test
+    public void spellCheckIsTrueWhenItsEnabled() {
+        spellCheckUrls = new String[]{"http://zcs.example.one:7780/aspell.php", " http://zcs.example.two:7780/aspell.php", "http://zcs.example.three:7780/aspell.php"};
+        setAttributes(spellCheckUrls);
+        GetInfo getInfo = new GetInfo();
+        boolean value = getInfo.isSpellCheckServiceAvailable(server);
+        Assert.assertEquals(true, value);
+    }
+
+    /**
+     * spellcheck should return false when zimbraSpellCheckURL on the current server returns an empty list of urls.
+     */
+    @Test
+    public void spellCheckIsTrueWhenItsEnabledFalse() {
+        setAttributes(spellCheckUrls);
+        GetInfo getInfo = new GetInfo();
+        boolean value = getInfo.isSpellCheckServiceAvailable(server);
+        Assert.assertEquals(false, value);
+    }
+
+    /**
+     * spellcheck should return true when zimbraSpellCheckURL on the current server returns non-empty list of urls.
+     * zimbraServiceInstalled and zimbraServiceEnabled shouldn't impact the return value of this method even "spell" service is not available on current server.
+     */
+    @Test
+    public void spellCheckIsFalseWhenItsNotEnabled() {
+        spellCheckUrls = new String[]{"http://zcs.example.one:7780/aspell.php", " http://zcs.example.two:7780/aspell.php", "http://zcs.example.three:7780/aspell.php"};
+        servicesEnabled = new String[]{ "amavis", "antivirus", "opendkim", "stats", "mailbox", "proxy"};
+        servicesInstalled = new String[]{"amavis", "antivirus", "opendkim", "stats", "mailbox", "proxy"};
+        setAttributes(servicesEnabled, servicesInstalled, spellCheckUrls);
+        GetInfo getInfo = new GetInfo();
+        boolean value = getInfo.isSpellCheckServiceAvailable(server);
+        Assert.assertEquals(true, value);
+    }
+
+    /**
+     * spellcheck should return false when zimbraSpellCheckURL on the current server returns with empty list of urls.
+     * zimbraServiceInstalled and zimbraServiceEnabled shouldn't impact the return value of this method even "spell" service is available on current server.
+     */
+    @Test
+    public void spellCheckIsTrueEvenWhenItsNotInstalled() {
```

Comment:
Isn't it "spellCheckIsFalseWhen..."?
True -> False?
Is "Even" necessary?
I don't know how "Enabled" or "Installed" is chosen for names of the functions.

Reply:
updated.